### ISSUE TITLE
Application bages (3rd party, deprecated, difficulty)

### ISSUE THREAD

**Issue by odinuv** [keboola/kbc-ui#1464]
Potrebujem u Appek *jasne* zobrazit, ze nejsou kebooli (jsou 3rd party) - cili asi nejaky badge na overview
Celkove musime v seznamu appek zobrazit:
- jestli je kebooli nebo nase (teoreticky mozna neco jako jako trusted developer jestli nekdy neco takovyho zavedeme)
- jestli je appka deprecated https://github.com/keboola/kbc-ui/issues/1062
- obtiznost konfigurace (asi 3 levely)

prisahal bych, ze uz to @jancermak  nekde kreslil, ale nemuzu to najit

**Comment by padak** [comment]
Je to všechno tady: https://keboola.productboard.com/feature-board/39535-jack-ux

**Comment by odinuv** [comment]
konkretne teda tady https://keboola.productboard.com/feature-board/39535-jack-ux/features/164421/detail

**Comment by padak** [comment]
Odlišit komponenty, co jsou skryté, aby bylo jasné, že pracuješ s DEVEL nebo PRIVATE komponentou. Na úrovni projektu je to v momentě použití úplně identické.

**Comment by odinuv** [comment]
ref https://keboola.productboard.com/feature-board/6271-planning/features/801694/detail

**Comment by janmichek** [comment]
Jdu do toho xD

**Comment by janmichek** [comment]
Rad bych tady shrnul dosavadni zadani, nez se do toho pustim

**Funkcionalita**
per komponentu v listu komponent chceme symbolizovat:

- platnost ['deprecated' + datum]
- puvod [badge: '3rd party']
- vlastnictvi [badge: 'private']
- slozitost nastaveni [badge: 'easy', 'normal', 'hard'] (zatim stejne neumime)

Pripada mi to teda dost informaci na kazdou komponentu, muze to byt dost vizualne tezky, tam zobrazit tolik info. Jsem pro nejakou minimalizaci - napr zobrazit jen badge puvodu. Alternativne muzem neco presunout jen do detailu komponenty.

**Vzhled**
Tady jsou posledni malovanky od Jacka:
https://keboola.productboard.com/feature-board/39535-jack-ux/features/164421/detail

Pravdepodobne to uz nevyhovuje zadani, tak se na to ve stredu s @jancermak  mrknem a zaktualizujem. Jsem osobne pro umisteni jinde nez v levem hornim rohu, protoze je to pravdepodone prvni vec co si 
 pak prectes.

**Postrehy**
VIzualne i obsahve mi to koliduje s timto seznamem v detailu komponenty = potreboval bych to nejdriv trochu rozebrat, zvalidovat s novymy navrhy a znovu slozit.

![image](https://user-images.githubusercontent.com/15363559/39423432-300a1e7c-4c72-11e8-8fce-078fdd25e8df.png)

- Predpokladam ze 'This is a 3rd party extractor' se automaticky rovna 'Support for this extractor is provided by its author, not Keboola'
- 'This extractor extracts data from outside sources' - heh, to je preci princip extractoru, ne? Nebo proc je tam takovy info?
- 'This extractor sends data outside of Keboola Connection' - tahle informace je pro me divna, moc mi toho nerika. Extractor ma preci sosat data, ne odesilat.

V listu komponent budeme chtit zobrazit symbol + max 2 slova. Nekde mozna bude potreba detailnejsi popis

**Comment by padak** [comment]
V přehledu komponenty bych dal jen 3rd a Beta. Ostatní nuže být v detailu

**Comment by janmichek** [comment]
Beta = Private , predpokladam.

**Comment by janmichek** [comment]
S Jackem dneska pripravime vizualni navrh.
Ve vypisu komponent teda chceme symbolizovat puvod -(label '3rd party'), vlastnictvi - (label 'Beta').
V detailu komponenty se potom zobrazi informace:

puvod [badge: '3rd party' + informace o supportu]
vlastnictvi [badge: 'private']
platnost ['deprecated' + datum]
slozitost nastaveni [badge: 'easy', 'normal', 'hard'] (zatim stejne neumime).
smer toku dat - u extractoru upozornit na dataout, u writeru na datain

mozna bude potreba trochu upravit texty a predstavoval bych si vizualni jednotu - detail, badge v seznamu

**Comment by janmichek** [comment]
v detailu se jeste muze objevit informace:

**Comment by odinuv** [comment]
> smer toku dat - u extractoru upozornit na dataout, u writeru na datain

pak jsou jeste aplikace u kterych to muze byt jakkoliv

**Comment by janmichek** [comment]
@odinuv  jenom chci checknout ze:

flag "excludeFromNewList" = private komponenta = label Beta -> OK?

**Comment by odinuv** [comment]
jo, ale spis nez Beta bych tomu rikal Alfa, nebo "Developer preview" nebo tak nejak. Jako Beta mame sami veci, ktery jsou public a celkem funkcni

**Comment by janmichek** [comment]
Budiz label 'Alfa' s pripadnym description 'developer preview'

**Comment by jancermak** [comment]
Ahoj — @janmichek @odinuv  — posilm navrh prehledu komponent a detailu.

**Návrh v invissionu:**
https://invis.io/NYHWJ68TQJC#/294829986_-1464_-_Application_Badges

**Navrh badgu:**
![image](https://user-images.githubusercontent.com/1826442/39537270-4f63a9aa-4e39-11e8-8d33-11a4ba89082b.png)

**SVG badge ke stazeni zde:**
[component-badges.zip](https://github.com/keboola/kbc-ui/files/1968195/component-badges.zip)

**Comment by janmichek** [comment]
![image](https://user-images.githubusercontent.com/15363559/39579756-cd86bfd0-4ee7-11e8-818e-ac1b3a1c3b78.png)
 
Tady mam zatim prubeznej stav. Badge se zatim zobrazujou takhle v gridu. Odstranil jsem teda 'filter by excluded' 

https://github.com/keboola/kbc-ui/pull/1660/commits/12b9325309e646db21b1122b5f893b77807779e7#diff-5b1f3afe00a24ee747357cf94f373324

takze ve vypisu se nyni zobrazi i private komponenty s badgem 'Alpha'.

Jedu dal, na detail. TAm se mi budou hodit nase SVG ikony z https://github.com/keboola/indigo-ui/issues/144

**Comment by janmichek** [comment]
V tom pripade me zajima jak poznam, kterou komponentu oznacit 'Private'

**Comment by janmichek** [comment]
Respektive 'Alpha'

**Comment by janmichek** [comment]
Oznacil jsem to podle tohoto https://github.com/keboola/kbc-ui/issues/1464#issuecomment-385936743

**Comment by odinuv** [comment]
no vsak private komponenty jsou ty co maji excludeFromNewList - a jak nazev napovida, tak nejsou videt v seznamu novych komponent, badge ale bude videt v detailu - o to jde hlavne. Dneska kdyz si tu komponentu nekdo nekonfiguruje, tak uz ti nikdo nijak nepozna, ze byla schovana

**Comment by janmichek** [comment]
aaaha, got it

**Comment by janmichek** [comment]
Odkladam si sem odkazy na komponenty, ktery maji hodne badgu:

/applications/fisa.app-gd-custom-colors
/applications/geneea.nlp-analysis-v2
/extractors/keboola.dark-sky-augmentation
/extractors/bizztreat.ex-fedger

**Comment by janmichek** [comment]
Dotazy na stredu pro Jacka @jancermak - probereme ve stredu
- ve vizualnim navrhu jsi zmenil border radius badgu - to se ma promitnout i na vsechny badge v aplikaci nebo jen na tyto?
- U tohoto by se hodilo vedle sebe videt buttony a labely (badge) jak jsem prudil uz driv. Mam pocit ze jsou prilis vizualne podobny (souvisi to i s dotazem vyse)

**Comment by odinuv** [comment]
@padak jeste me napadlo - nechtels v detailu komponenty zobrazovat, ze ma pristup k internetu? (nebo staci dataIn/dataOut ? )

**Comment by padak** [comment]
na to bych teď asi prděl - je to jasný z kontextu (nejde tolik o přístup k
internetu, ale jaký data z KBC umí dostat)

**Comment by jancermak** [comment]
@janmichek  
![image](https://user-images.githubusercontent.com/1826442/39805671-d47c5428-5377-11e8-803d-e6ae73c21d9f.png)

Nahodil jsem ti jak vedle sebe vypada badge (label) a button. BTN ma radius 3px a badge label je pulkruh, tedy radius je 1/2 vysky.

Vysel jsem z pripravovaneho style guidu — kde je mam takto pouzite v tabulkach.
![image](https://user-images.githubusercontent.com/1826442/39805815-2590d4ce-5378-11e8-8f06-b3d09f59e9a4.png)

**Comment by janmichek** [comment]
Odkladam si jeste je je potreba upravit Guide Mode - odebral jsem tlacitko 'more' v gridu komponent

**Comment by janmichek** [comment]
Po dokonceni featury jsem udelal debriefing z Issue a PR.
Padly tam nasledujici veci, ale nikdo tomu nevenoval pozornost, tak nechavam na posouzeni, jestli se tomu jeste venovat:

- otagovat dle slozitosti nastaveni
- logika renderovani - rozdeli na 2 komponenty -pro detail a Novou komponentu

**Comment by Halama** [comment]
Uplně jsem nesledoval diskuzi kolem toho ani jsem do toho nechtěl kecat. Ale přijde mi že to dost chybí na seznamu konfigurací komponent:
![image](https://user-images.githubusercontent.com/903531/40606535-f6d90c26-6265-11e8-9c1d-9af75001acf5.png)

**Comment by janmichek** [comment]
Jj, diky. Uz mi jsme zazanemanli a pripisuju.

**Comment by odinuv** [comment]
@padak je to v procesu https://keboola.slack.com/archives/C02KU1V9S/p1527522124000330
